src/services/big-query-client.ts

```diff
@@ -0,0 +1,41 @@
+import { ResultWithoutExpectedErrors } from '../doppler-types';
+import { ResponseCache } from './../utils';
+import { GenericClient } from './generic-client';
+
+export interface EmailList {
+  emails: String[];
+}
+
+export interface BigQueryClient {
+  getEmailsData(idUser: String): Promise<ResultWithoutExpectedErrors<EmailList>>;
+}
+
+export class HttpBigQueryClient extends GenericClient implements BigQueryClient {
+  private responseCache = new ResponseCache();
+
+  private mapEmails(response: any): EmailList {
+    return {
+      emails: response.emails,
+    };
+  }
+  public async getEmailsData(idUser: String): Promise<ResultWithoutExpectedErrors<EmailList>> {
+    try {
+      const { jwtToken } = super.getApiConnectionData();
+      const response = await this.axios.request({
+        method: 'GET',
+        url: `/big-query/emails`,
```

Comment:
The URL should include the accountname (email) because it helps a lot in the security implementation of our APIs:

```suggestion
      const { email, jwtToken } = super.getApiConnectionData();
      const response = await this.axios.request({
        method: 'GET',
        url: `/big-query/{email}/emails`,
```

Reply:
all apis methods must always have that url param?

Reply:
if the result is related to data of an specific account, yes.

We can talk about the reason.

Reply:
the email was incorporated in the query string

Reply:
Thanks!

Just a minor correction, you have incorporated the email in the path:

```
          userinfo       host      port
          ┌──┴───┐ ┌──────┴──────┐ ┌┴┐
  https://john.doe@www.example.com:123/forum/questions/?tag=networking&order=newest#top
  └─┬─┘   └───────────┬──────────────┘└───────┬───────┘ └───────────┬─────────────┘ └┬┘
  scheme          authority                  path                 query           fragment
```